sound/soc/intel/common/soc-acpi-intel-cml-match.c

```diff
@@ -9,45 +9,52 @@
 #include <sound/soc-acpi.h>
 #include <sound/soc-acpi-intel-match.h>
 
-static struct snd_soc_acpi_codecs cml_codecs = {
+static struct snd_soc_acpi_codecs rt1011_spk_codecs = {
 	.num_codecs = 1,
-	.codecs = {"10EC5682"}
+	.codecs = {"10EC1011"}
 };
 
-static struct snd_soc_acpi_codecs cml_spk_codecs = {
+static struct snd_soc_acpi_codecs max98357a_spk_codecs = {
 	.num_codecs = 1,
 	.codecs = {"MX98357A"}
 };
 
+/*
+ * The order of the three items with .id = "10EC5682" matters
+ * here, because the ACPI ID of max98357a is wrongly present
+ * in the board with rt1011 speaker codec.
```

Comment:
because the DSDT tables expose an ACPI HID for the MAX98357A speaker amplifier which is not populated on the board.